### PR TITLE
epic: esl-animate

### PR DESCRIPTION
**Pull request by ala-n**
Initial implementation of the esl-animate module.

Closes: #646

### CODE REVIEW

File: src/modules/esl-animate/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Features:
+- Add class(es) when observed elements enter viewport area
+- Support group animation that allows item delay its animation on the passed time after previously animated item
+- Support forward and backward animations directions
+- Support automatic re-animate after item exit viewport area
+- JS API (Service) + Custom Element - Plugin for simple initialization
+
+## ESLAnimateService
+ESLAnimateService - is a core of esl-animate module. Element is needs to be observed by ESLAnimateService
```
Comment (julia-murashko) [inline src/modules/esl-animate/README.md:21]:
```suggestion
ESLAnimateService - is a core of esl-animate module. Element needs to be observed by ESLAnimateService 
```

File: src/modules/esl-animate/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- JS API (Service) + Custom Element - Plugin for simple initialization
+
+## ESLAnimateService
+ESLAnimateService - is a core of esl-animate module. Element is needs to be observed by ESLAnimateService 
+in order to be animated.
+
+### Service API:
+- `ESLAnimateService.observe(els, config)` - method to start element observation
+  - `els` - Element or array of Elements to observe and animate
+  - `config` - optional ESLAnimateConfig object to describe animate functionality behaviour
```
Comment (julia-murashko) [inline src/modules/esl-animate/README.md:27]:
```suggestion
  - `config` - an optional ESLAnimateConfig object to describe the behavior of the animation functionality
```

File: src/modules/esl-animate/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- `ESLAnimateService.unobserve(els)` - method to unsubscribe ESLAnimateService from elements observation
```
Comment (julia-murashko) [inline src/modules/esl-animate/README.md:28]:
```suggestion
- `ESLAnimateService.unobserve(els)` - method to unsubscribe ESLAnimateService from observing elements
```

**Review by julia-murashko — APPROVED** [review]

File: src/modules/esl-animate/core/esl-animate-service.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  /** Intersection observable callback */
+  @bind
+  protected onIntersect(entries: IntersectionObserverEntry[]): void {
+    entries.forEach(({target, intersectionRatio, isIntersecting}: IntersectionObserverEntry) => {
+      const config = this.getConfigFor(target);
+      if (!config) return;
+
+      if (intersectionRatio >= 0.4) this._entries.add(target);
```
Comment (dshovchko) [inline src/modules/esl-animate/core/esl-animate-service.ts:97]: Perhaps it makes sense to define values via class properties? For ease of inheritance and overriding.

**Review by dshovchko — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-animate/core/esl-animate-service.ts (hunk 4 above)
Comment (ala-n) [reply src/modules/esl-animate/core/esl-animate-service.ts]: Updated, thanks for the suggestion 👍🏻

**Review by NastaLeo — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

**Pull request merged**